src/librbd/exclusive_lock/PostAcquireRequest.cc

```diff
@@ -172,11 +173,52 @@ void PostAcquireRequest<I>::handle_allocate_journal_tag(int r) {
     return;
   }
 
+  send_open_image_cache();
+}
+
+template <typename I>
+void PostAcquireRequest<I>::send_open_image_cache() {
+  CephContext *cct = m_image_ctx.cct;
+  ldout(cct, 10) << dendl;
+
+  using klass = PostAcquireRequest<I>;
+  Context *ctx = create_async_context_callback(
+    m_image_ctx, create_context_callback<
+    klass, &klass::handle_open_image_cache>(this));
+  cache::rwl::InitRequest<I> *req = cache::rwl::InitRequest<I>::create(
+    m_image_ctx, ctx);
+  req->send();
+}
+
+template <typename I>
+void PostAcquireRequest<I>::handle_open_image_cache(int r) {
+  CephContext *cct = m_image_ctx.cct;
+  ldout(cct, 10) << "r=" << r << dendl;
+
+  save_result(r);
+  if (r < 0) {
+    lderr(cct) << "failed to open image cache: " << cpp_strerror(r)
+               << dendl;
+    send_close_journal();
+    return;
+  }
+  // With RWL cache, discard granularity_bytes = 0
+  if (m_image_ctx.image_cache && m_image_ctx.discard_granularity_bytes) {
+    ldout(cct, 1) << "RWL image cache is enabled and "
+                  << "set discard_granularity_bytes = 0." << dendl;
+    m_image_ctx.discard_granularity_bytes = 0;
```

Comment:
Nit: probably should live in the `rwl::InitRequest` state machine

Reply:
Moved